timemachine/fe/rbfe.py

```diff
@@ -0,0 +1,423 @@
+import functools
+import io
+import warnings
+from collections import namedtuple
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pymbar
+from rdkit import Chem
+from rdkit.Chem import AllChem, Draw
+
+from timemachine.constants import BOLTZ, DEFAULT_TEMP
+from timemachine.fe.single_topology_v3 import SingleTopologyV3
+from timemachine.fe.system import convert_bps_into_system
+from timemachine.fe.utils import get_romol_conf
+from timemachine.ff.handlers import openmm_deserializer
+from timemachine.lib import LangevinIntegrator, MonteCarloBarostat, custom_ops
+from timemachine.md import minimizer
+from timemachine.md.barostat.utils import get_bond_list, get_group_indices
+
+
+def plot_atom_mapping_grid(mol_a, mol_b, core_smarts, core, show_idxs=False):
```

Comment:
Should this be moved into a plotting utilities module or similar?

(This file contains the only library code with a dependence on matlotlib. (Other usages appear in `attic`, `tests`, or commented-out code.) Implementation also seems equally applicable to other cases where atom-mapping is used, such as relative hydration.)

Reply:
This plot is specific to a relative free energy calculation though - not sure what the best way to organize this is.

Reply:
Should matplotlib be added as a dependency then?

Reply:
good call - hm

Reply:
added matplotlib as a dependency. I'd like to remove this at a future deboggling though